velox/expression/CastExpr.cpp

```diff
@@ -103,10 +124,22 @@ VectorPtr CastExpr::castToDate(
   switch (fromType->kind()) {
     case TypeKind::VARCHAR: {
       auto* inputVector = input.as<SimpleVector<StringView>>();
+      const auto& queryConfig = context.execCtx()->queryCtx()->queryConfig();
+      auto isStandardCast = queryConfig.isIso8601();
       applyToSelectedNoThrowLocal(context, rows, castResult, [&](int row) {
         try {
           auto inputString = inputVector->valueAt(row);
-          resultFlatVector->set(row, DATE()->toDays(inputString));
+          auto date = util::castFromDateString(inputString, isStandardCast);
+          if (date.has_value()) {
+            resultFlatVector->set(row, *date);
+          } else {
+            if (setNullInResultAtError()) {
+              resultFlatVector->setNull(row, true);
```

Comment:
we dont need to handle this case here since applyToSelectedNoThrowLocal() will take care of setting the errors for the respective rows and there is common logic to handle setting nulls on error.

Reply:
Apart from the case where nullptr is returned, there are also a case in castFromDateString() that return a velox user error (daysSinceEpochFromDate) still which needs to be handled in a consistent way. The previous patchset that returned a VELOX_USER_ERROR should suffice